superset/views/database/api.py

```diff
@@ -274,6 +351,139 @@ def table_metadata(self, database: Database, table_name: str, schema_name: str):
+          - in: path
+            name: pk
+            description: The database id
+            schema:
+              type: integer
+          - in: path
+            description: Table schema
+            name: schema
+            schema:
+              type: string
+          - in: path
+            description: Sub string
+            name: substr
+            schema:
+              type: string
+          - in: path
+            description: force cache refresh
+            name: force_refresh
+            schema:
+              type: bool
+          responses:
+            200:
+              description: Table schema info
+              content:
+                text/plain:
+                  schema:
+                    type: object
+                    properties:
+                      options:
+                        type: array
+                        items:
+                          type: object
+                          properties:
+                            label:
+                              type: string
+                            schema:
+                              type: string
+                            title:
+                              type: string
+                            type:
+                              type: string
+                            value:
+                              type: string
+                      tableLength:
+                        type: number
+            400:
+              $ref: '#/components/responses/400'
+            401:
+              $ref: '#/components/responses/401'
+            404:
+              $ref: '#/components/responses/404'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        self.incr_stats("init", self.tables.__name__)
+        force_refresh_parsed = force_refresh.lower() == "true"
+        schema_parsed = parse_js_uri_path_item(schema_name, eval_undefined=True)
+        substr_parsed = parse_js_uri_path_item(substr, eval_undefined=True)
+        # Guarantees filtering on databases
+        database: Database = self.datamodel.get(pk, self._base_filters)
+        if not database:
+            self.incr_stats("error", self.tables.__name__)
+            return self.response_404()
+
+        tables = get_all_table_names_in_database(
+            database, schema_parsed, substr_parsed, force_refresh_parsed
+        )
+        views = get_all_view_names_in_database(
+            database, schema_parsed, substr_parsed, force_refresh_parsed
+        )
+
+        if not schema_parsed and database.default_schemas:
+            user_schema = g.user.email.split("@")[0]
+            valid_schemas = set(database.default_schemas + [user_schema])
+
+            tables = [tn for tn in tables if tn.schema in valid_schemas]
+            views = [vn for vn in views if vn.schema in valid_schemas]
+
+        max_items = app.config["MAX_TABLE_NAMES"] or len(tables)
```

Comment:
Is there a way we can get at this configuration value without pulling in the entire `app` object? Is it possible to use the `current_app` proxy or load the config value directly?

Reply:
Changed it to `current_app`